processor/src/main/java/org/mapstruct/ap/internal/model/ValueMappingMethod.java

```diff
@@ -485,16 +500,28 @@ String getValue(ValueMappingOptions valueMapping) {
     }
 
     private ValueMappingMethod(Method method, List<MappingEntry> enumMappings, String nullTarget, String defaultTarget,
-        boolean throwIllegalArgumentException, List<LifecycleCallbackMethodReference> beforeMappingMethods,
+        Type defaultException,
+        List<LifecycleCallbackMethodReference> beforeMappingMethods,
```

Comment:
I know it is not part of this solution.. but I'm not happy with those methods coming from `@BeanMapping`.. I think they should be provided by `@EnumMapping`

Reply:
the selectors are coming from `BeanMapping` indeed. I think that this is a good think for the 2.0 API overhauling. perhaps the qualifiers should go in their own annotation for this.